src/ring_theory/ideal/operations.lean

```diff
@@ -95,6 +95,21 @@ begin
   exact Hb _ hi _ hj,
 end
 
+/-- Dependent version of smul_induction_on -/
+@[elab_as_eliminator]
+theorem smul_induction_on' {x : M} (hx : x ∈ I • N)
```

Comment:
I think this works better if x and hx are the last arguments; does that match `span_induction` or other similar dependent versions? Or is that a bad comparison?

Reply:
It matched `smul_induction_on` from which I started to work, but `submodule.span_induction` has `x` and `hx` as last arguments.
I can try to rewrite them in the same way, but it could take time to check
that it does not break anything.